itest/assets_test.go

```diff
@@ -334,3 +343,170 @@ func testMintAssetNameCollisionError(t *harnessTest) {
 	collideAssetName := rpcCollideAsset[0].AssetGenesis.Name
 	require.Equal(t.t, commonAssetName, collideAssetName)
 }
+
+func testMintAssetsWithTapscriptSibling(t *harnessTest) {
+	ctxb := context.Background()
+	ctxt, cancel := context.WithTimeout(ctxb, defaultWaitTimeout)
+	defer cancel()
+
+	aliceUtxoResp := t.lndHarness.Alice.RPC.ListUnspent(
+		&walletrpc.ListUnspentRequest{},
+	)
+	aliceUtxoCount := len(aliceUtxoResp.Utxos)
+
+	bobUtxoResp := t.lndHarness.Bob.RPC.ListUnspent(
+		&walletrpc.ListUnspentRequest{},
+	)
+	bobUtxoCount := len(bobUtxoResp.Utxos)
+
+	t.Logf("initial UTXO counts: %d, %d", aliceUtxoCount, bobUtxoCount)
+
+	// Build the tapscript tree.
+	sigLockPrivKey := test.RandPrivKey(t.t)
+	hashLockPreimage := []byte("foobar")
+	hashLockLeaf := test.ScriptHashLock(t.t, hashLockPreimage)
+	sigLeaf := test.ScriptSchnorrSig(t.t, sigLockPrivKey.PubKey())
+	siblingTree := txscript.AssembleTaprootScriptTree(hashLockLeaf, sigLeaf)
+
+	siblingBranch := txscript.NewTapBranch(
+		siblingTree.RootNode.Left(), siblingTree.RootNode.Right(),
+	)
+	typedBranch := asset.TapTreeNodesFromBranch(siblingBranch)
+	innerBranch := asset.GetBranch(typedBranch).UnwrapToPtr()
+	rawBranch := asset.ToBranch(*innerBranch)
+	siblingReq := mintrpc.FinalizeBatchRequest_Branch{
+		Branch: &taprpc.TapBranch{
+			LeftTaphash:  rawBranch[0],
+			RightTaphash: rawBranch[1],
+		},
+	}
+
+	rpcSimpleAssets := MintAssetsConfirmBatch(
+		t.t, t.lndHarness.Miner.Client, t.tapd, simpleAssets,
+		WithSiblingBranch(siblingReq),
+	)
+	rpcIssuableAssets := MintAssetsConfirmBatch(
+		t.t, t.lndHarness.Miner.Client, t.tapd, issuableAssets,
+	)
+
+	AssertAssetBalances(t.t, t.tapd, rpcSimpleAssets, rpcIssuableAssets)
+	// assert on taproot_asset_root, merkle_root, and internal_key
+	// + assets have chain_anchor struct with tapscript_sibling
+
+	utxos, err := t.tapd.ListUtxos(ctxt, &taprpc.ListUtxosRequest{})
+	require.NoError(t.t, err)
+	utxosJSON, err := formatProtoJSON(utxos)
+	require.NoError(t.t, err)
+
+	t.Logf("UTXOs: %v", utxosJSON)
+
+	// Filter the managed UTXOs to select the genesis UTXO with the
+	// tapscript sibling.
+	utxoWithSibling := func(utxo *taprpc.ManagedUtxo) bool {
+		return !bytes.Equal(utxo.TaprootAssetRoot, utxo.MerkleRoot)
+	}
+	mintingOutputWithSibling := fn.Filter(
+		maps.Values(utxos.ManagedUtxos), utxoWithSibling,
+	)
+	require.Len(t.t, mintingOutputWithSibling, 1)
+	genesisWithSibling := mintingOutputWithSibling[0]
+
+	// Extract the fields needed to construct a script path spend.
+	mintTapTweak := genesisWithSibling.MerkleRoot
```

Comment:
Should we also re-compute the merkle root here and make sure it's sibling root + asset commitment root?